test/e2e/operator_test.go

```diff
@@ -1199,6 +1201,140 @@ func TestAWSLoadBalancerControllerOpenAPIValidation(t *testing.T) {
+	if err := waitForDeploymentStatusCondition(context.TODO(), t, kubeClient, defaultTimeout, deploymentName, expected...); err != nil {
+		t.Fatalf("did not get expected available condition for deployment: %v", err)
+	}
+
+	dep := &appsv1.Deployment{}
+	if err := kubeClient.Get(context.TODO(), deploymentName, dep); err != nil {
+		t.Fatalf("failed to get deployment %s: %v", deploymentName.Name, err)
+	}
+	depGeneration := dep.Generation
+
+	// Save a copy of the original infra Config, to revert changes before exiting.
+	originalInfra := infra.DeepCopy()
+	defer func() {
+		err := updateInfrastructureConfigStatusWithRetryOnConflict(t, 5*time.Minute, kubeClient, func(infra *configv1.Infrastructure) *configv1.Infrastructure {
+			infra.Status = originalInfra.Status
+			return infra
+		})
+		if err != nil {
+			t.Logf("Unable to remove changes from the infraConfig, possible corruption of test environment: %v", err)
+		}
+	}()
+
+	// Update infrastructure status with initialInfraTags
+	initialInfraTags := []configv1.AWSResourceTag{
+		{Key: "plat-key1", Value: "plat-value1"},
+		{Key: "conflict-key1", Value: "plat-value2"},
+		{Key: "conflict-key2", Value: "plat-value3"},
+	}
+	t.Logf("Updating AWS ResourceTags in the cluster infrastructure config: %v", initialInfraTags)
+	err = updateInfrastructureConfigStatusWithRetryOnConflict(t, 5*time.Minute, kubeClient, func(infra *configv1.Infrastructure) *configv1.Infrastructure {
+		if infra.Status.PlatformStatus == nil {
+			infra.Status.PlatformStatus = &configv1.PlatformStatus{}
+		}
+		if infra.Status.PlatformStatus.AWS == nil {
+			infra.Status.PlatformStatus.AWS = &configv1.AWSPlatformStatus{}
+		}
+		infra.Status.PlatformStatus.AWS.ResourceTags = initialInfraTags
+		return infra
+	})
+	if err != nil {
+		t.Errorf("failed to update infrastructure status: %v", err)
+	}
+
+	// wait for the deployment to restart and become available
+	if err := waitForDeploymentRollout(context.TODO(), t, kubeClient, defaultTimeout, deploymentName, depGeneration); err != nil {
+		t.Fatalf("deployment did not roll out within timeout: %v", err)
+	}
+	if err := waitForDeploymentStatusCondition(context.TODO(), t, kubeClient, defaultTimeout, deploymentName, expected...); err != nil {
+		t.Fatalf("did not get expected available condition for deployment: %v", err)
+	}
+
+	// get the updated deployment
+	if err := kubeClient.Get(context.TODO(), deploymentName, dep); err != nil {
+		t.Fatalf("failed to get deployment %s: %v", deploymentName.Name, err)
+	}
+	depGeneration = dep.Generation
+
+	// Check `--default-tags` arg for tags present in alb instance and infra status (initialInfraTags)
+	expectedTagValue := "conflict-key1=op-value2,conflict-key2=op-value3,op-key1=op-value1,plat-key1=plat-value1"
+	assertContainerArgFromDeployment(t, dep, awsLoadBalancerControllerContainerName, "--default-tags", expectedTagValue)
```

Comment:
I was thinking differently. The task of the ALBO is to set the flags correctly, next it's the job of ALBC to consume those flags and take necessary actions (adding tags in this case).  If we can validate that the `--default-tags` flag is getting populated correctly, I hope we can trust ALBC to propagate them to a load balancer. 

I would expect ALBC's e2e to query the tags using https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/elasticloadbalancingv2

In case of CIO also, we did a similar check. We're checking whether the operator has correctly set the annotations or not. https://github.com/openshift/cluster-ingress-operator/pull/1148/files#diff-cf4b4e5424070a666a364d1d1b04011478d888d6d3d65c943ca7783333b7a6e4R1356-R1358

Reply:
> The task of the ALBO is to set the flags correctly, next it's the job of ALBC to consume those flags and take necessary actions (adding tags in this case). If we can validate that the --default-tags flag is getting populated correctly, I hope we can trust ALBC to propagate them to a load balancer.

We don't have much control over what's validated upstream. The e2e tests for our addon operators verify use cases we think are nominal for our customers. And the idea here is to go from one end (API) to the other end (load balancers). That's why all the other e2e tests send real traffic to check the load balancers are in the shape we expect them to be. The validation of the expected values for `--default-tags` flag is what a unit test can do even without `envtest`, a fake Kube client would be enough.

> I would expect ALBC's e2e to query the tags using https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/elasticloadbalancingv2

I would expect too. But I don't think we know this for sure.

> In case of CIO also, we did a similar check. We're checking whether the operator has correctly set the annotations or not. https://github.com/openshift/cluster-ingress-operator/pull/1148/files#diff-cf4b4e5424070a666a364d1d1b04011478d888d6d3d65c943ca7783333b7a6e4R1356-R1358

C-I-O case is different. The creation of load balancers for `Service`s is out of control of C-I-O, it's not done by anything C-I-O manages or being responsible for.

Frankly speaking, I would be ok if this feature would come with a set of unit tests only. However since you already into the e2e stage, we should be consistent with the rest of the tests and go to very end (AWS).

Reply:
> Frankly speaking, I would be ok if this feature would come with a set of unit tests only. However since you already into the e2e stage, we should be consistent with the rest of the tests and go to very end (AWS).

Okay, that sounds reasonable to me. I've updated the e2e test to query tags from the ELB instance and asserted it based on the expected tags. Hope it addressed your suggestion and made it consistent with other tests.

Reply:
Running the e2e multiple times is causing Rate limiting error. I do have 10 seconds of delay before making any client calls, but that is also not enough, it seems. Could this be an issue in future? 
```
api error Throttling: Rate exceeded
```

Reply:
As discussed during the sync call, [the resource group tagging api](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/resourcegroupstaggingapi) can be tried to overcome the rate limiting.

Reply:
What I've checked is that, [the resource group tagging api](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/resourcegroupstaggingapi) will give us the list of loadbalancer ARNs given the tags. However, I think we still need to match the retrieved ARNs with the hostname/DNS name, which requires another API call (`DescribeLoadBalancers`). 

If you think that getting an ARN based on the expected tags is what we need, and we don't need to validate that ARN with the hostname, then it's worth giving a try. 

```go
input := &resourcegroupstaggingapi.GetResourcesInput{
		ResourceTypeFilters: []string{"elasticloadbalancing:loadbalancer"},
		TagFilters:          tagFilters,
	}

	var arns []string
	paginator := resourcegroupstaggingapi.NewGetResourcesPaginator(client, input)
	for paginator.HasMorePages() {
		page, err := paginator.NextPage(context.Background())
		if err != nil {
			return nil, fmt.Errorf("getting resources page: %w", err)
		}

		for _, resource := range page.ResourceTagMappingList {
			arns = append(arns, *resource.ResourceARN)
		}

	}
```

Reply:
As per the discussion, there should exist strictly a `1` load balancer since we're controlling the tags in the e2e tests and also resetting everything at the end of each test.